### PR TITLE
[23.05] nixos/gitlab: use git 2.41 for gitlab 16.3

### PR DESCRIPTION
**Pull request by yu-re-ka**
## Description of changes

Re-introduce the gitlab 16.3 upgrade, and use a new git version.
#253405 #253631 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/gitlab.nix [hunk 1]
```diff
@@ -14,6 +14,20 @@ let
                       else
                         pkgs.postgresql_13;
 
+  # gitlab 16.3.0 requires git 2.41.0, but nixos 23.05 is at 2.40.1
+  gitPackage = if
```
Comment (ajs124) [inline nixos/modules/services/misc/gitlab.nix:18]: why not use https://gitlab.com/gitlab-org/git/-/tags/v2.41.0.gl1?

**Comment by yayayayaka** [comment]
Result of `nixpkgs-review pr 253629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>gitaly</li>
    <li>gitlab</li>
    <li>gitlab-container-registry</li>
    <li>gitlab-ee</li>
    <li>gitlab-elasticsearch-indexer</li>
    <li>gitlab-pages</li>
    <li>gitlab-shell</li>
    <li>gitlab-workhorse</li>
  </ul>
</details>

File: nixos/modules/services/misc/gitlab.nix (hunk 1 above)
Comment (yayayayaka) [reply nixos/modules/services/misc/gitlab.nix:18]: It doesn't compile

**Review by yayayayaka — APPROVED** [review]
Successfully tested on x86_64-linux: git operations are working again.

**Pull request merged**